pkg/controller/migplan/registry.go

```diff
@@ -106,6 +120,11 @@ func (r ReconcileMigPlan) ensureRegistrySecret(client k8sclient.Client, plan *mi
 	if plan.EqualsRegistrySecret(newSecret, foundSecret) {
 		return foundSecret, nil
 	}
+	// secret is not same, we need to redeploy
+	deleteErr := r.deleteImageRegistryResourcesForClient(client, plan)
```

Comment:
I'm wondering whether we should just delete the registryDC here. The very next call below calls UpdateRegistrySecret with the updated content, so deleting it here is unnecessary. Also, I'm not quite sure what happens if you delete something and then make an update call, now that it no longer exists.

Reply:
+1, did not intend to delete the secret, I'll refactor. However, bouncing the pods here is necessary, right?